components/trade-slip/TradeSlipItem.tsx

```diff
@@ -0,0 +1,158 @@
+import { isIndexedData, isNA } from "@zeitgeistpm/sdk-next";
+import { Decimal } from "decimal.js";
+import { ZTG } from "lib/constants";
+import { usePrevious } from "lib/hooks/usePrevious";
+import { TradeSlipItem, useTradeslipItems } from "lib/state/tradeslip/items";
+import {
+  itemKey,
+  useTradeslipItemsState,
+  useTradeslipItemState,
+} from "lib/state/tradeslip/tradeslipItemsState";
+import { useStore } from "lib/stores/Store";
+import { observer } from "mobx-react";
+import { FC, useEffect, useState } from "react";
+import { X } from "react-feather";
+import { useDebounce } from "use-debounce";
+import { AmountInput } from "../ui/inputs";
+
+export type TradeSlipItemProps = {
+  item: TradeSlipItem;
+  disabled?: boolean;
+};
+
+const TradeSlipItem = observer<FC<TradeSlipItemProps>>(({ item, disabled }) => {
+  const { config } = useStore();
+
+  const { put, removeAsset } = useTradeslipItems();
+  const state = useTradeslipItemState(item);
+
+  const [amountInput, setAmountInput] = useState(item.amount);
+  const [debouncedAmountInput] = useDebounce(amountInput, 300);
+  const prevDebouncedAmountInput = usePrevious(debouncedAmountInput);
+
+  useEffect(() => {
+    if (prevDebouncedAmountInput !== debouncedAmountInput) {
+      put({ ...item, amount: Number(debouncedAmountInput) });
```

Comment:
I would prefer more verbose but meaningful function names

Reply:
For the put function?

Reply:
Yeah

Reply:
I disagree. The tradeslip items atom is just storage and put is a pretty conventional method name that does either create or update depending on if the item exists. Which is what you want in a context where you dont know if you are creating the item or updating it. https://restfulapi.net/rest-put-vs-post/